### ISSUE TITLE
Support for Jekyll 4.0

### ISSUE THREAD

**Issue by janko** [github/pages-gem#651]
[Jekyll 4.0.0 is out](https://jekyllrb.com/news/2019/08/20/jekyll-4-0-0-released/), and brings lots of nice improvements. It would be really nice if `github-pages` gem would add support for Jekyll 4.

**Comment by fsimonis** [comment]
Noteworthy is the upgrade of `jekyl-sass-converter` to version 2.
They switched from `Ruby Sass` to `csass`, as the former has __reached end-of-life__:

```
Post-install message from sass:

Ruby Sass has reached end-of-life and should no longer be used.

* If you use Sass as a command-line tool, we recommend using Dart Sass, the new
  primary implementation: https://sass-lang.com/install

* If you use Sass as a plug-in for a Ruby web framework, we recommend using the
  sassc gem: https://github.com/sass/sassc-ruby#readme

* For more details, please refer to the Sass blog:
  https://sass-lang.com/blog/posts/7828841
```

**Comment by polyrain** [comment]
Can we get an update on this?

**Comment by wigging** [comment]
I am also awaiting Jekyll v4.0 support for Github Pages. I need to use the "manually order documents" feature for collections but it is only available in v4.0. https://jekyllrb.com/docs/collections/#manually-ordering-documents

**Comment by DirtyF** [comment]
I'd recommend using  GitHub Actions to build your site and publish it on GitHub Pages: https://github.com/BryanSchuetz/jekyll-deploy-gh-pages

**Comment by kushalpandya** [comment]
I didn't have any dependency on Jekyll 4.0 except for better SCSS compile performance (due to sassc), but I ditched GitHub Pages in favour of Netlify as it can even install 3rd-party Jekyll plugins which aren't supported by GitHub, and you get all the features that GitHub Pages provides + a few extra for free!

**Comment by DirtyF** [comment]
Please note that Netlify recently limited their free plan to 300min build per month, aka 10min per day.

**Comment by wigging** [comment]
In the past, how long did it take for GitHub Pages to update to the latest version of Jekyll? Or is GitHub Pages always on an old Jekyll version?

**Comment by kushalpandya** [comment]
> Please note that Netlify recently limited their free plan to 300min build per month, aka 10min per day.

I think 10 min per day is sufficient for most users who are using GitHub Pages where they don't update their site as often each day. In my case, I have a moderately complex site with 50+ documents and a handful of 3rd party jekyll plugins and yet the site builds gets published under 40 seconds on Netlify. So hitting that 10 minute limit for the day would be distant. 😅 

> In the past, how long did it take for GitHub Pages to update to the latest version of Jekyll? Or is GitHub Pages always on an old Jekyll version?

I think upgrade to 4.0 is going to take some time as it being a major upgrade with 1 breaking change around how `post_url` behaves as GitHub wouldn't want a lot of broken static sites with such an upgrade. Not to mention that many repos which use GH Pages don't have `Gemfile` present (myself included when I used Jekyll for the first time to build my portfolio) which is going to affect them with upgrade.

**Comment by brando90** [comment]
what is the progress for this? Is there support for Jekyll 4?

**Comment by kimchirichie** [comment]
As a new user to Jekyll, i am unsure how to post to github pages without the 4.0 support. Without this library, all my links break due to relative paths. (or maybe im ignorant). Would anyone have any suggestions on which previous version i should be using?  My bundler says 
```
 github-pages was resolved to 4, which depends on
      jekyll (= 1.1.2)
```
which doesnt seem right. Regardless, I've given that a try by lowering  `jekyll@v1.1.2` which is when I ran into issues with `jekyll-feed` not being supported at `jekyll@v1.1.2`. So I removed that and got
```
An error occurred while installing yajl-ruby (1.1.0), and Bundler cannot continue.
Make sure that `gem install yajl-ruby -v '1.1.0' --source 'https://rubygems.org/'` succeeds before bundling.
```
Ugh... It would be nice to use `jekyll@v4.0` but for the time being, which version should I use?  Is there a branch that supports `jekyll@v4.0`?

**Comment by kimchirichie** [comment]
after a bit of hair pulling, I ACTUALLY read the comments in the `Gemfile` and commented out the gem `jekyll` and uncommented `github-pages` as instructed. hopefully someone finds my findings useful down the road.
```
# gem "jekyll", "~> 4.0.0"
gem "github-pages", group: :jekyll_plugins
group :jekyll_plugins do
  gem "jekyll-feed", "~> 0.11"
end
```

**Comment by bguiz** [comment]
Jekyll 4 has [added support for binary operators within `where_exp`](https://github.com/jekyll/jekyll/pull/6998), and I would like to use this feature. Can we please get a release of `github-pages` that works with Jekyll 4 pls?

BTW @kimchirichie thanks for [this comment](https://github.com/github/pages-gem/issues/651#issuecomment-555890008) - that works out on localhost, but unfortunately not on our particular CI/CD setup - so still need a new release of this page.

**Comment by hoff2-ACN** [comment]
Not sure if this is any help but I suspect one could do a jekyll 4 build in a github action now. Seems plausible enough to look into whether someone's already done it, in fact. Been trying to make time to give it a shot myself.

**Comment by sujaykundu777** [comment]
@hoff2-ACN I have built my site using Github Pages and Github Actions with the latest Bundler and Jekyll 4.0.0 out of the box, soon will be sharing the tutorial which can be used to deploy your Blog using Jekyll 4 !! 👍

**Comment by hoff2-ACN** [comment]
@sujaykundu777 fantastic! thank you so much!

**Comment by sujaykundu777** [comment]
@hoff2-ACN  Here it is 🍡  - [Deploy Jeyll Blog using Github Actions with Jekyll 4](https://sujaykundu.com/blog/post/deploy-jekyll-using-github-pages-and-github-actions#/)

**Comment by jdickey** [comment]
So, with people either going to GitHub Actions or to Netlify, are we to assume that GitHub no longer cares about the traditional write/commit/push-and-publish workflow? This issue has been open well over five months, and four since a committer to this repo (@dirtyf) left [a comment](#issuecomment-541442887).

There are clearly a number of people waiting/hoping that one of the very few who are familiar with the codebase (if I'm reading [this chart](https://github.com/github/pages-gem/graphs/contributors) right, about 3 people) can either perform the (apparently non-trivial) updates needed to close this issue *or* open a PR with enough information for one or more of us tyros to get started.

Are we waiting for Godot?

**Comment by DirtyF** [comment]
_Disclaimer: I don't work for GitHub, nor am a maintainer for this gem._ 

The update is indeed non-trivial, as there are breaking changes and some impacting issues were raised since Jekyll 4.0 was released, due to the new cache API, so **it's safer for GitHub not to upgrade to Jekyll 4.0 for now, as they don't want to break existing websites.** 

As this was already mentioned before, **GitHub Actions are the way to go** if you want to use a custom Jekyll or plugins at your own risk. 

I don't have proper rights on this repository, but this issue should be closed for now, as it won't happen anytime soon, no need to give people false hopes. GitHub should come with a proper documentation on how to run Jekyll on GitHub Pages with GitHub Actions.

**Comment by jdickey** [comment]
Sorry, @DirtyF, I misunderstood your relationship to the project (and possibly GitHub) from the quality of the comments you'd left earlier.

Actions *sans* this Gem sound like The Way Forward for those of us who'd like to take advantage of Jekyll 4 (and future versions). Thanks again.

**Comment by DirtyF** [comment]
On a side note; `remote-theme` plugin is now supoprting Jekyll 4.0 : https://github.com/benbalter/jekyll-remote-theme/releases/tag/v0.4.2

**Comment by jasongaylord** [comment]
Any update on the support of Jekyll 4?

**Comment by sujaykundu777** [comment]
For reference you can checkout this theme which uses Jekyll 4.0 :  [devlopr-jekyll](https://github.com/sujaykundu777/devlopr-jekyll ) 🍡

**Comment by jasongaylord** [comment]
> For reference you can checkout this theme which uses Jekyll 4.0 : [devlopr-jekyll](https://github.com/sujaykundu777/devlopr-jekyll) 🍡

And how does this theme support version 4 on GitHub Pages? The GitHub Pages Gem only supports 3.8.

**Comment by DirtyF** [comment]
Jekyll team released a guide on how to use Jekyll 4 and custom pluginsq on GitHub Pages with GitHub Actions: https://jekyllrb.com/docs/continuous-integration/github-actions/

**Comment by sujaykundu777** [comment]
> > For reference you can checkout this theme which uses Jekyll 4.0 : [devlopr-jekyll](https://github.com/sujaykundu777/devlopr-jekyll) 🍡
> 
> And how does this theme support version 4 on GitHub Pages? The GitHub Pages Gem only supports 3.8.

It uses GitHub Action for using Jekyll 4 to build and deploy your site automatically to gh-pages branch.

**Comment by jasongaylord** [comment]
Thanks @DirtyF. That's helpful. I did not know they had a supported method.

It would still be nice if GitHub updated their gem to support it.

**Comment by blakegearin** [comment]
Jekyll `3.9.0` and lower contains the high severity vulnerability [CVE-2020-14001](https://github.com/advisories/GHSA-mqm2-cgpr-p4m6). It's not present in Jekyll `4.0.0` or higher. This alone makes it unadvisable to use this gem if you care about the security of your project; it's not simply a nicety.

Please provide an update on when this will be addressed

**Comment by ashmaroli** [comment]
@blakegearin Jekyll 3.9.0 doesn't *contain the said vulnerability*. It is flexible enough to allow users upgrade on their discretion. Users just need to run `gem update kramdown` or `bundle update kramdown`.

**Comment by blakegearin** [comment]
@ashmaroli Thanks, I wasn't aware of that! For reference I was able to do that successfully and confirm the vulnerability wasn't present 🙂

**Comment by wigging** [comment]
Still waiting on GitHub Pages to support Jekyll 4 which is currently sitting at version 4.1.1

**Comment by DirtyF** [comment]
Jekyll 4.0 has breaking changes and that would break millions of Jekyll sites on Github, so either stick with 3.9.0 or [use GitHub Actions to deploy the latest Jekyll 4.x on GitHub Pages](https://jekyllrb.com/docs/continuous-integration/github-actions/)

**Comment by riuson** [comment]
@DirtyF, we are have already file `.nojekyll`. Why not addd some like `.jekyll4` to specify not default version?

**Comment by limansky** [comment]
I think the solution might be a new gem with another name, like `github-pages4` providing Jekyll 4 and friends. This would allow people to use new Jekyll without playing with any kind of CI.

**Comment by wigging** [comment]
GitHub Actions is not a solution for Jekyll 4 support.

**Comment by parkr** [comment]
@wigging GitHub Actions could build your site and push the resulting compiled site to the `gh-pages` branch each time it ran. This is how TravisCI and other CI providers have offered GitHub Pages building in the past. Can you explain why this is not a solution? It would allow you to control your environment and build automatically.

**Comment by wigging** [comment]
@parkr I don't consider that a solution, it's more of a workaround to the current problem. GitHub Pages currently uses the old version 3.9 of Jekyll where the user can just push changes to GitHub to update their Jekyll site. No GitHub Actions or CI needed. I want that same workflow for the latest version of Jekyll. Is GitHub Pages going to be stuck on Jekyll 3.9 while the rest of the world moves on to Jekyll 4, 5, 6, and beyond?

**Comment by garrett** [comment]
FWIW: I'm part of a team where we have a website auto-update on push.

We each have our own forks for previewing. It's easy to opt-into the standard GitHub pages auto-building. It's not easy for each of us to set up actions or CI for our forks.

I imagine a lot of other folks are working in a similar manner.

It would be great if GitHub could finally, officially update GitHub Pages to Jekyll 4.x, or at least communicate what's going on and if there's an ETA. (4.0 was released over a year ago.)

**Comment by cseas** [comment]
Any update on when support for Jekyll 4.0+ might be added? My project has started throwing deprecation warnings which look like they won't away unless I upgrade Jekyll 3.x to 4.x (https://github.com/jekyll/jekyll/issues/7947)

**Comment by andrewbanchich** [comment]
@parkr My Jekyll themes use link tags, which behave differently in 4.0. This breaks things for people using GitHub Pages, but not for people deploying with GitHub Actions / building somewhere else.

Can we please update this so I don't have to choose between users I support?

**Comment by andrewbanchich** [comment]
By choosing to not update Jekyll, GitHub is breaking all new sites going forward.

Regarding breaking existing sites, I expect that would only happen if users push to update their site again.

For those who do push, couldn't GitHub do something to prevent breaking sites anyway? E.g. introduce some custom config file for GH Pages that allows you to specify the Jekyll version, and refuse to compile site updates that don't include that file?

**Comment by alejandro5042** [comment]
> E.g. introduce some custom config file for GH Pages that allows you to specify the Jekyll version, and refuse to compile site updates that don't include that file?

Or perhaps, assume that the Jekyll version is the current one (as of the time of this comment). So if you want Jekyll 4+, you'll have to include this file.

Bonus points if this config isn't in its own file but part of an existing file, e.g. the Gemfile.

**Comment by jdickey** [comment]
The fact that this discussion is still ongoing a year on is...concerning. Have GitHub written off supporting Jekyll going forward? If not, credibility would be significantly enhanced by either forward motion wrt Pages/Jekyll or a migration path forward. Just IMO.

**Comment by chicio** [comment]
@nat please can owe have an answer on this topic? 🙏

**Comment by wigging** [comment]
Jekyll is now at v4.2 (https://jekyllrb.com). By the time GitHub supports v4.0, Jekyll will already be at v5.0. 🤷

**Comment by twsl** [comment]
Any updates? Can't we just merge #745 ?

**Comment by wigging** [comment]
It's been more than 1.5 years since Jekyll 4.0 came out. I don't think GitHub will ever update GitHub Pages to support the latest Jekyll version.

**Comment by runxel** [comment]
> It's been more than 1.5 years since Jekyll 4.0 came out. I don't think GitHub will ever update GitHub Pages to support the latest Jekyll version.

Which in itself would not be dramatic, but the lack of recognition and official answers here is.
Bad, bad communication. 👎

**Comment by br3nt** [comment]
Who wants Jekyll 4?? We want Jekyll 4!! When I say Jekyll, you say 4! Jekyll!! <Github Team> 3!

**Comment by alantbarlow** [comment]
I need Jekyll 4 support to prioritize includes over excludes in the config file. It's been 2 years since its release and no one has even been assigned to this issue.

**Comment by yaitskov** [comment]
Please remove constraint on gem versions.
You wont be able to keep up.

**Comment by hansfn** [comment]
As suggested earlier adding support for the current version of Jekyll (without breaking all existing sites on Github Pages), could be as easy as adding a file `.useCurrentJekyll` (similar to `.nojekyll`) to enable it.

Anyway, an official reply from Github would be nice. I tried to make some noise on Twitter 😆

**Comment by janderssonse** [comment]
I wrote on their latest changelog post on linkedin and asked about attention to this issue. :)

**Comment by brutallyminimal** [comment]
This works very well for me for deploying Jekyll 4+ - https://jekyllrb.com/docs/continuous-integration/github-actions/

**Comment by hh-lohmann** [comment]
DirtyF said on Feb 1, 2020 what has to be said (see above, shortcut https://github.com/github/pages-gem/issues/651#issuecomment-581069671), so close this and other threads by appropriate information, eg. "We at GitHub regard Jekyll 4.x not as an update of Jekyll, but as a separate program, that, as many separate programs, is supported by Actions". Jekyll was always, reasonably, restricted at GitHub, maybe they should **officially** label a "GitHub Flavored Jekyll", like "GitHub Flavored Markdown" that had, eg., no footnotes until recently (https://github.blog/changelog/2021-09-30-footnotes-now-supported-in-markdown-fields/).

(btw if you didn't know: https://jekyllrb.com/news/2021/09/14/goodbye-dear-frank/ (=DirtyF))

**Comment by AntonyLeons** [comment]
The solution here is to remove the github pages gem and add any missing dependacies by hand. Then you can ise a CI/CD service like netlify or upload your html pages from your build folder. 

old GemFile:
```
source "https://rubygems.org"

gem "jekyll", "~> 3.8.5"
gem 'jekyll-feed'
gem 'github-pages', '~> 206'
gem 'rake', '~> 13.0'
gem 'wdm', '~> 0.1.1'
```
new Gemfile:
```
source "https://rubygems.org"

gem 'jekyll', '~> 4.2', '>= 4.2.1'
gem 'jekyll-paginate', '~> 1.1'
gem 'jekyll-feed', '~> 0.16.0'
gem 'jekyll-sitemap', '~> 1.4'
gem 'rake', '~> 13.0', '>= 13.0.6'
gem 'wdm', '~> 0.1.1'
```

This is just pushing you to use a service like github actions though, probably why they refuse to update this.

**Comment by pensierinmusica** [comment]
@parkr @benbalter @yoannchaudet @felipesu19 any chance GitHub pages upgrade Jekyll >= 4.0 ?

It's been a long time now, and there are features needed.

**Comment by yoannchaudet** [comment]
We are looking at ways to support Jekyll 4+ but not this way.

I promise you exciting things are coming. We just need a little bit more time on our side (a few weeks).

**Comment by jlosito** [comment]
I believe at this point I'm starting to see transient dependency vulnerabilities due to the version of jekyll =3.9.2 which v227 requires, latest at the time of writing.

**Comment by yoannchaudet** [comment]
@wigging We release on 🗓 7/27! I am making a note to post back here once it's out :)

**Comment by garrett** [comment]
@yoannchaudet: Firstly: Congrats!

---

Sadly, I'm not sure how to adapt to this new workflow.

For Cockpit's website (https://cockpit-project.org/ with our repo @ https://github.com/cockpit-project/cockpit-project.github.io), we use a model where people fork the site and use their fork as a preview of what's going to be in the PR.

Since this GitHub Pages change is quite a bit different... I don't know how that would work exactly, without having to have everyone (not just our own repo members, but also contributors) manually set up their own configuration, which would be a big hassle. Right now, it's as simple as visiting the settings once, going to "Pages", setting a branch and clicking "save"... just once and then they're done.

(For example, when we have a release, we open a PR, such as https://github.com/cockpit-project/cockpit-project.github.io/pull/549. And then we link to our own personal fork as a preview, which in this case is https://garrett.github.io/cockpit-project.github.io/blog/cockpit-273.html)

---

Is there a way where we can have a configuration file that's in the main repo that can be reused without modification for preview forks that does all the necessary setup? If so, this could possibly even make it easier.

Thanks, again!

**Comment by chrispat** [comment]
@garrett once you have added the workflow to your repo all repos will get that workflow so all they will need to do is enable pages in their fork.  At that point it should work the same way.

**Comment by mattxwang** [comment]
Great news @yoannchaudet, congratulations!

Could I ask - from a theme maintainer's perspective, this is very exciting news (and something that we've been waiting for)! Do you know what criterion you're evaluating before the feature becomes stable (ex out of public beta), or a roadmap/timeline? We're currently thinking about whether or not we want to drop Jekyll 3 support, and this feature is our last hurdle to do so.

**Comment by philippart-s** [comment]
@yoannchaudet thanks for the news about the new custom github actions workflow for pages.
Is it meaning that pages-gem will not be updated to support Jekyll 4 and to have this support we must use the new github actions workflow ?

**Comment by yoannchaudet** [comment]
No there is no immediate plants to add Jekyll 4 to this gem. Doing so would require some extensive work in the gem or on the GitHub side to ensure we don't break the millions of Pages sites we build (or have built) with Jekyll 3 up until today.

**Comment by soundasleep** [comment]
> No there is no immediate plants to add Jekyll 4 to this gem. Doing so would require some extensive work in the gem or on the GitHub side to ensure we don't break the millions of Pages sites we build (or have built) with Jekyll 3 up until today.

Would it be possible to opt-in to Jekyll 4, perhaps? Creating a `.jekyll-version` file with `4.0` in the root?

**Comment by fenixnet-net** [comment]
3.5 years later....

**Comment by wigging** [comment]
I gave up using Jekyll for GitHub Pages since GitHub refuses to support the latest Jekyll version. So I made my own static site generator called genja (https://github.com/wigging/genja). It is a small Python package that you can run from the terminal. Thought I would share it here in case others want to use it. Good luck to everyone else who still uses Jekyll. I'm moving on. Bye 👋

**Comment by jlosito** [comment]
@fenixnet-net @wigging everyone is able to build Jekyll sites that aren't stuck using GitHub's gem anymore using the action workflow linked below.

https://github.com/actions/starter-workflows/blob/main/pages/jekyll-gh-pages.yml

**Comment by wigging** [comment]
@jlosito Yes I know. I don't want to setup a GitHub Action just to build a website.

**Comment by wigging** [comment]
@limansky Yes, I just build the site locally (on my laptop) and push up the files to the `docs` folder in the repository. I do a lot of Python programming so using a Python tool is much easier and convenient for me. Since genja is a Python package, I could setup a GitHub Action (CI) to build the site, but it's easier for me to do all that locally. I can also preview the site locally before pushing up the changes. I could do all these things with Jekyll but I had to use an older version of Jekyll for GitHub Pages. Also, I don't do anything in Ruby so it didn't make sense to setup Ruby just to make a Jekyll website.

**Comment by fenixnet-net** [comment]
@jlosito As @wigging points out, one shouldn't have to set up an entire custom workaround just to *use* a service github allegedly provides.

Jeckyll 4 came out 3.5 years ago. If github was interested in keeping Pages relevant they'd have long since at least published a timeline for implementing the upgrade. To my eyes it looks like this feature is now going to be left to die on the vine, as it were. It's reminiscent of Google Voice - yeah, it's still working, but it's at this point a zombie on a treadmill, shambling forward without going anywhere.

**Comment by 0xdevalias** [comment]
> I could do all these things with Jekyll but I had to use an older version of Jekyll for GitHub Pages.

@wigging Not really, you could use any local version of jekyll you wanted in the same 'build locally then deploy' method that you're now using for your home-grown tool. You just needed to add a .nojekyll file when uploading your assets:

- https://github.blog/2009-12-29-bypassing-jekyll-on-github-pages/
- https://github.com/0xdevalias/devalias.net/blob/master/_config.yml#L106-L110

---

> If github was interested in keeping Pages relevant they'd have long since at least published a timeline for implementing the upgrade

@fenixnet-net They basically have, the modern supported method is 'use the GitHub action'; I suspect they will eventually convert the current legacy system to just be some syntactic sugar on top of GitHub actions, or remove it completely.

**Comment by TWiStErRob** [comment]
> I suspect they will eventually convert the current legacy system to just be some syntactic sugar on top of GitHub actions - @0xdevalias [comment](https://github.com/github/pages-gem/issues/651#issuecomment-1484466273)

<details><summary>I think they already did it!</summary>

![image](https://github.com/github/pages-gem/assets/2906988/f515dc85-79f2-4d6b-8974-18ec9cb47235)
![image](https://github.com/github/pages-gem/assets/2906988/f8c20f00-1118-43ca-8362-1993c3755f5d)
![image](https://github.com/github/pages-gem/assets/2906988/e367e28a-0663-433d-b9ac-c3093a57a343)

</details>

**Comment by xepozz** [comment]
Any updates?

**Comment by chicio** [comment]
I suppose GitHub/Microsoft don't have any interest in this (and I can understand why, there are a lot of alternatives in the market)

**Comment by wigging** [comment]
I gave up on using Jekyll for GitHub Pages and made my own static site generator.

**Comment by TWiStErRob** [comment]
@wigging and others, what is the reason for being against Jekyll in general? As far as I know Jekyll is not affiliated with GitHub/Microsoft (not even [sponsoring](https://github.com/jekyll/jekyll?tab=readme-ov-file#sponsors)), GitHub Pages is just using an otherwise popular technology.

I think we need to distinguish between `pages-gem` abandoning major updates that follow Jekyll (this issue) vs. Jekyll being a good community maintained tool independent of GitHub pages, that happens to be quite easily hosted on GitHub pages just like any other site generator.

**Comment by wigging** [comment]
When I first learned about GitHub Pages, I read the page at https://pages.github.com which links to Jekyll for blogging. So that is why I started using Jekyll, because I was interested in blogging with GitHub Pages. As I learned more about Jekyll I began to build the site locally so I could preview it before pushing changes to GitHub. I eventually ran into issues because my local Jekyll would be newer than the supported Jekyll version on GitHub Pages. I started using Jekyll features that were not supported on GitHub Pages. That is why I stopped using Jekyll because I didn't want to downgrade my local version just to make it compatible with GitHub Pages. I also don't like Ruby and I didn't want to maintain a Ruby environment just for Jekyll. Anyway, after all of that, it was easier for me to create my own static site generator instead of using Jekyll or some other third-party solution.

**Comment by 0xdevalias** [comment]
> I think we need to distinguish between `pages-gem` abandoning major updates that follow Jekyll (this issue) vs. Jekyll being a good community maintained tool independent of GitHub pages, that happens to be quite easily hosted on GitHub pages just like any other site generator.

And probably a 3rd option to add to that nuance, which is the now canonical (since ~2022) GitHub actions based workflow that more or less renders `pages-gem` irrelevant these days.

I'm honestly confused as to what people are complaining about in this issue these days.. there is ample documentation about how to use the modern version of Jekyll on GitHub pages, and as they stated in their release blog for that feature, there is no one canonical standard version/setup of Jekyll/plugins/etc, so it really doesn't make sense to try and update this legacy gem to support something that no longer exists.

You could go back and read this comment where it was announced in this very thread back in July 2022: https://github.com/github/pages-gem/issues/651#issuecomment-1197448026

Or here, where I pointed it out again in March 2023: https://github.com/github/pages-gem/issues/651#issuecomment-1484466273

Or this part in October 2023 when what I said was confirmed: https://github.com/github/pages-gem/issues/651#issuecomment-1773912726

And seemingly that was enough for most people until ~15 hours ago when we got a bunch of new comments on this issue, which at best guess I imagine mustn't have actually bothered to read any of the past information that would have given them all of the answers they needed.

So.. once again.. here is more information than you should ever need about this issue, and why it's irrelevant to spam it (and everyone who was following it) with comments such as "any updates" or "I abandoned because.." or other similar unhelpful comments:

- https://github.blog/changelog/2022-07-27-github-pages-custom-github-actions-workflows-beta/
  - > GitHub Pages: Custom GitHub Actions Workflows (beta)
  - > July 27, 2022
- https://github.blog/news-insights/product-news/github-pages-now-uses-actions-by-default/
  - > GitHub Pages now uses Actions by default
  - > August 10, 2022
  - > As Jekyll grew in popularity, many different plugins were added, providing greater capability. Eventually a version of Jekyll was released that wasn’t compatible with some of the core plugins provided by the GitHub Pages platform. The pagesworker that powered the build and deployment of pages sites was a single-purpose system that didn’t support versioning, which left us unable to upgrade to new Jekyll versions or easily add more plugins without breaking users.
    >
    > Furthermore, as time has passed, many other site generators have been created and you have asked us to support them. By bringing the power of GitHub Actions to the GitHub Pages platform, you have the freedom to build your static website using any framework that exists today or is created in the future.
  - Which even explicitly talks about how to use Jekyll 4+, including a starter workflow
    - https://github.blog/news-insights/product-news/github-pages-now-uses-actions-by-default/#how-can-i-upgrade-to-jekyll-4
      - > ### How can I upgrade to Jekyll 4+?
        > 
        > For that, you will want to switch from your GitHub Pages source from “branch” to “GitHub Actions” (see the [changelog](https://github.blog/changelog/2022-07-27-github-pages-custom-github-actions-workflows-beta/)) and use a custom workflow. We published a [starter workflow](https://github.com/actions/starter-workflows/blob/main/pages/jekyll.yml) you can leverage today. It will require you to have a `Gemfile` and optionally a configuration file in your repository. You can follow [Jekyll’s getting started guide](http://jekyllrb.com/docs/).
- https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site
  - > GitHub Pages now uses GitHub Actions to execute the Jekyll build. When using a branch as the source of your build, GitHub Actions must be enabled in your repository if you want to use the built-in Jekyll workflow. Alternatively, if GitHub Actions is unavailable or disabled, adding a `.nojekyll` file to the root of your source branch will bypass the Jekyll build process and deploy the content directly.
- https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site
  - > Configuring a publishing source for your GitHub Pages site
    > You can configure your GitHub Pages site to publish when changes are pushed to a specific branch, or you can write a GitHub Actions workflow to publish your site.
  - https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-with-a-custom-github-actions-workflow
    - > Publishing with a custom GitHub Actions workflow
- https://docs.github.com/en/pages/getting-started-with-github-pages/using-custom-workflows-with-github-pages
  - > Using custom workflows with GitHub Pages
    > You can take advantage of using GitHub Actions and GitHub Pages by creating a workflow file or choosing from the predefined workflows.
  - https://github.com/actions/configure-pages
    - > configure-pages
      > A GitHub Action to enable Pages and extract various metadata about a site. It can also be used to configure various static site generators we support as [starter workflows](https://github.com/actions/starter-workflows/tree/main/pages).
  - https://github.com/actions/upload-pages-artifact
    - > upload-pages-artifact
      > A composite Action for packaging and uploading artifact that can be deployed to GitHub Pages.
- https://github.com/actions/deploy-pages
  - > deploy-pages
  - > This action is used to deploy Actions artifacts to GitHub Pages.
- https://github.com/actions/starter-workflows/tree/main/pages
  - https://github.com/actions/starter-workflows/blob/main/pages/jekyll.yml
    - > This workflow uses actions that are not certified by GitHub. They are provided by a third-party and are governed by separate terms of service, privacy policy, and support documentation.
    - > Sample workflow for building and deploying a Jekyll site to GitHub Pages
    - > `name: Deploy Jekyll site to Pages`
  - https://github.com/actions/starter-workflows/blob/main/pages/jekyll-gh-pages.yml
    - > Sample workflow for building and deploying a Jekyll site to GitHub Pages
    - > `name: Deploy Jekyll with GitHub Pages dependencies preinstalled`

---

So.. after having read all of that.. and understanding what the current situation is for Jekyll on GitHub pages.. does anyone have any legitimate concerns to be raised here anymore? Or can we move on from this again now like it seemed we had concluded back in 2023?

---

**Edit:** See also, this issue that I just opened to have all references to the legacy `github-pages` gem removed from the GitHub documentation:

- https://github.com/github/docs/issues/36740

**Comment by wigging** [comment]
I support closing this issue. Let's move on.

**Comment by 0xdevalias** [comment]
> See also, this issue that I just opened to have all references to the legacy `github-pages` gem removed from the GitHub documentation

Here's the latest on this; if anyone has thoughts on how best the docs should be updated, including which pages/etc, please head over to https://github.com/github/docs/issues/36740 and leave your thoughts there:

> I finally got the go-ahead from both the docs and pages teams, so we can discuss the kind of note you think would work best. My instinct is a call-out at the beginning of the article you've mentioned here along the lines of "Jekyll is no longer being updated, and Actions is better for most use-cases," with a link to whatever "about Actions" or quickstart article we have. But that's just initial thoughts, I'm definitely open to other options. Are there any other articles you know of that would benefit from the same treatment? 
> 
> _Originally posted by @Sharra-writes in https://github.com/github/docs/issues/36740#issuecomment-3001836427_

**Comment by 0xdevalias** [comment]
Updates from:

- https://github.com/github/docs/issues/36740

> Okay! I've added a note to all the relevant Jekyll docs I could find, confirmed that information about implementing Jekyll with Actions and the `.nojekyll` file is available on our site (even if you probably have to use Google to find it without knowing the file name), and cleaned up the "who can use this feature" box since no one was reading it anyway. The note is also a reusable, so if anyone finds another article that could use the warning, it can easily be added. _Hopefully_ all this makes your lives a little easier going forward. 🤞
>
> _Originally posted by @Sharra-writes in https://github.com/github/docs/issues/36740#issuecomment-3145426737_

PR of that change for future reference:

- https://github.com/github/docs/pull/39148

Example of a page that includes it:

- https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll
  - <img width="710" height="114" alt="Image" src="https://github.com/user-attachments/assets/42835f91-9360-4a7d-86bb-1b350b37106a" />
